### ISSUE TITLE
Feature request: Make Sorting by last modified configurable as default

### ISSUE THREAD

**Issue by michaelaye** [jupyter/notebook#1267]
I prefer to have my notebooks sorted by "last modified" timestamp, so I'd like to be able to have this sorting as default, after launching a new notebook server.
Having this available could possibly solve #1218 as well, maybe?

**Comment by takluyver** [comment]
This seems like a good use case for the front-end config mechanism - rather
than finding a config option to set the default, it could just remember
what sort order you used last.

**Comment by michaelaye** [comment]
is that available currently?

**Comment by takluyver** [comment]
The mechanism to store and load config from the frontend is, though we don't have great docs for it. Here's what we do have:

http://jupyter-notebook.readthedocs.org/en/latest/frontend_config.html
https://github.com/jupyter/scipy-advanced-tutorial/blob/master/Part1/06-storing-config.md

**Comment by captainsafia** [comment]
Working on this now. Woot! No docs! Wild, wild West!

**Comment by minrk** [comment]
@captainsafia if you make a PR with persisting the option and dev docs for the relevant config API, you get :cake:.

**Comment by chuaxr** [comment]
Seconding this request, any help would be appreciated.

**Comment by michaelaye** [comment]
I think it’s solved by #2641 ?

**Comment by chuaxr** [comment]
@michaelaye would you elaborate on the version you are using, and the behavior you are referencing? I upgraded from 5.0 to 5.1 using conda. In both cases, I am able to change the sorting option on one page, but this ordering does not persist if I enter a folder or open the tree from a new window.

**Comment by michaelaye** [comment]
Yeah I didn’t say it’s released. :) it’s in GH master but not released yet.

**Comment by takluyver** [comment]
#2641 is released as part of 5.1, but I don't think that goes as far as what you want.

#2641 aims to ensure that we don't randomly reset the sort order while looking at one folder in one page, but it doesn't persist the sort order. So when you go to another folder, or open the file list in another tab, you will see the default sort order. Persisting it would require another PR.

**Comment by chuaxr** [comment]
@takluyver should I create a separate feature request?

**Comment by takluyver** [comment]
I think this issue is the feature request. It's still open.

**Comment by superhonya** [comment]
any update on this issue?
it'd make a nice improvement for sure

**Comment by JozefK78** [comment]
indeed. it would be nice, if people didn't have to start every session by this step.

**Comment by slagtermaarten** [comment]
I second this request :)

**Comment by erjan** [comment]
i too ask for this lil thing to be done. it's just additional click on "last modified"  and it wastes time

**Comment by captainsafia** [comment]
I've unassigned myself from this since I likely won't be getting to this any time soon (and unfortunately in the past as well). However, I'm happy to guide folks through an implementation.

Implementing this feature will require familiarity with JavaScript in general and jQuery in particular.  The implementation will live in [this file](https://github.com/jupyter/notebook/blob/43df5af2b614088b4b297fae90a70b6505b9bf84/notebook/static/tree/js/notebooklist.js), which contains the code for interactivity on the tree view of files.

This file defines a set of [sort functions](https://github.com/jupyter/notebook/blob/43df5af2b614088b4b297fae90a70b6505b9bf84/notebook/static/tree/js/notebooklist.js#L89) for sorting by different attributes. Each sort function is uniquely identified by a key (e.g. `last-modified`). We'll probably want people to configure this in a file.

These mapping is used in the [sort_list](https://github.com/jupyter/notebook/blob/43df5af2b614088b4b297fae90a70b6505b9bf84/notebook/static/tree/js/notebooklist.js#L269) function which is used in the event callback for the click action on the sort button ([here](https://github.com/jupyter/notebook/blob/43df5af2b614088b4b297fae90a70b6505b9bf84/notebook/static/tree/js/notebooklist.js#L249)). The type of sort used is dependent on the button clicked.

I imagine the implementation will exist somewhere in the [load_list](https://github.com/jupyter/notebook/blob/43df5af2b614088b4b297fae90a70b6505b9bf84/notebook/static/tree/js/notebooklist.js#L375), which renders the list for the first time. It will have to invoke the `sort_list` function and pass the key reference for the last-modified attribute.

**Comment by castorfou** [comment]
I second that request too ;) Thanks to @captainsafia's guidance it may be possible to do it. Even for me?

**Comment by tfiers** [comment]
Just to add, for discoverability: there's [this comment](https://github.com/jupyter/notebook/issues/184#issuecomment-593328879) on an old issue where someone asked for the same thing, with a workaround by @MarkMoretto just below it (via a `custom.js` that click the sort button after a timeout).
(Found via [this SO comment](https://stackoverflow.com/questions/31142810/sort-sorting-order-of-files-in-notebook-list#comment121072176_63318546) on the same issue).
